### PR TITLE
Add runtime page

### PR DESCRIPTION
**Pull request by bulislaw**
I'm not sure what else could go on this page, most of the runtime info will be in the subpages.

@sg- @0xc0170 @AnotherButler

### CODE REVIEW

**Comment by AnotherButler** [comment]
Query: Which page(s) deal with memory initialization, ctors and dtors? I don't believe I've seen that information yet.

**Comment by bulislaw** [comment]
There's a passage about standard library initialisation.

File: docs/reference/runtime/runtime.md [hunk 1]
```diff
@@ -1,2 +1,13 @@
...
+To start developing applications using Mbed OS, you need to create a main function and start coding. In most cases, no special setup is necessary.
+
+Write your Mbed OS applications in C++ (C++98).
+
+### Compilers
+
+Mbed OS supports:
+
+- GNU Arm Embedded Toolchain (GCC) version 6.
+- Arm Compiler version 5 and 6.
```
Comment (0xc0170) [inline docs/reference/runtime/runtime.md]: I am not certain listing both here. As currently 5 is supported (ARM6 is there but fails for various targets and missing prebuilt libraries that needs updating). 

I would assume ARM5 (ARM6 experimental) currently, in 5.7 only ARM6.

File: docs/reference/runtime/runtime.md [hunk 2]
```diff
@@ -1,2 +1,13 @@
 ## Runtime overview
 
+To start developing applications using Mbed OS, you need to create a main function and start coding. In most cases, no special setup is necessary.
+
+Write your Mbed OS applications in C++ (C++98).
+
+### Compilers
```
[the remark this answers is not in the document]
Comment (bulislaw) [reply docs/reference/runtime/runtime.md]: I would say that should be in one place, but the Jira ticket mentioned versions.

File: docs/reference/runtime/runtime.md [hunk 3]
```diff
@@ -1,2 +1,13 @@
 ## Runtime overview
 
+To start developing applications using Mbed OS, you need to create a main function and start coding. In most cases, no special setup is necessary.
```
[the remark this answers is not in the document]
Comment (bulislaw) [reply docs/reference/runtime/runtime.md]: I would say not here, we do mention it across the docs, when we talk about specifics (like the main stack size configuration).

**Comment by AnotherButler** [comment]
@sg- Please weigh in.

**Comment by bulislaw** [comment]
As in get rid of this page?

**Comment by sg-** [comment]
Just put on ice for now and come back to later.

**Pull request closed without merging**